### PR TITLE
Reorganize scripts and use new javascript ivar to insert them into the document

### PR DESCRIPTION
**Pull request by fanxhe-pair**

### CODE REVIEW

File: lib/livingstyleguide/templates/scripts/search.html.erb [hunk 1]
```diff
@@ -0,0 +1,67 @@
+<script>
```
Comment (hagenburger) [inline lib/livingstyleguide/templates/scripts/search.html.erb]: Why not removing this `<script>` HTML tag and renaming the file to *search.js.erb?*

File: lib/livingstyleguide/filters/search_box.rb [hunk 2]
```diff
@@ -1,9 +1,10 @@
 LivingStyleGuide.default_options[:search_box] = {
-  placeholder: "Search …"
+  placeholder: "Search …"
 }
 
 LivingStyleGuide.add_filter :search_box do |arguments, options, content|
   placeholder = options[:placeholder] || LivingStyleGuide.default_options[:search_box][:placeholder]
-  document.header << ERB.new(File.read("#{File.dirname(__FILE__)}/../templates/search-box.html.erb")).result(binding)
```
Comment (hagenburger) [inline lib/livingstyleguide/filters/search_box.rb]: For consistency with the other files in that folder, I would keep the dash.

File: lib/livingstyleguide/document.rb [hunk 3]
```diff
@@ -84,10 +84,10 @@ def erb
   end
 
   def evaluate(scope, locals, &block)
+    scripts_path = "#{File.dirname(__FILE__)}/templates/scripts"
     @head = ""
-    @head << ERB.new(File.read("#{File.dirname(__FILE__)}/templates/javascript-copy.html.erb")).result(binding)
-    @head << ERB.new(File.read("#{File.dirname(__FILE__)}/templates/code-example.html.erb")).result(binding)
-    @head << ERB.new(File.read("#{File.dirname(__FILE__)}/templates/colors-example.html.erb")).result(binding)
+    @javascript = ""
+    @javascript << ERB.new(File.read("#{File.dirname(__FILE__)}/templates/scripts.html.erb")).result(binding)
```
Comment (hagenburger) [inline lib/livingstyleguide/document.rb]: Just directly read and add the files here. No need to call an ERB template for this as there’s nothing else than Ruby code.

File: lib/livingstyleguide/templates/layout.html.erb [hunk 4]
```diff
@@ -11,6 +11,7 @@
 // see: http://www.hagenburger.net/BLOG/Simple-HTML5-Fix-for-IE.html
 for(var e,l='article aside footer header nav section time picture'.split(' ');e=l.pop();document.createElement(e));
 </script>
+<%= javascript %>
```
Comment (hagenburger) [inline lib/livingstyleguide/templates/layout.html.erb]: Move one line up to be inside of the `<script>` tag.

File: lib/livingstyleguide/templates/scripts/search.html.erb [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    while (j >= 0) {
+      addMatch(matched, selection[j]);
+      if (hasClass(selection[j], "lsg--headline")) {
+        break;
+      }
+      j--;
+    }
+  }
+  document.addEventListener("DOMContentLoaded", function() {
+
```
Comment (hagenburger) [inline lib/livingstyleguide/templates/scripts/search.html.erb]: Switch line 35 and 36 for more beautiful code.

**Comment by fanxhe-pair** [comment]
Ping @hagenburger already updated :)

**Pull request merged**